src/core/mesh/qgsmeshdataprovider.cpp

```diff
@@ -48,14 +48,14 @@ QgsMeshDriverMetadata QgsMeshDataProvider::driverMetadata() const { return QgsMe
 
 QgsMeshDatasetIndex QgsMeshDatasetSourceInterface::datasetIndexAtTime(
   const QDateTime &referenceTime,
-  int groupIndex, quint64 time,
+  int groupIndex, qint64 time,
   QgsMeshDataProviderTemporalCapabilities::MatchingTemporalDatasetMethod method ) const
 {
   const QDateTime requestDateTime = referenceTime.addMSecs( time );
-  quint64 providerTime;
+  qint64 providerTime;
   const QDateTime providerReferenceTime = mTemporalCapabilities->referenceTime();
   if ( mTemporalCapabilities->referenceTime().isValid() )
-    providerTime = referenceTime.msecsTo( requestDateTime );
+    providerTime = providerReferenceTime.msecsTo( requestDateTime );
```

Comment:
how it thiis related to the bug? Is is separate bug? how it was found?

Reply:
I saw this code error reading it.
It is necessary when there is a gap between reference time between settings and provider.
But fix it leads to a failure of a test related to files with inconsistent reference time...
The error fixed here was hiding the inconsistent reference time in test files...

So now, it is too tricky to fix it here, I move this fix in another PR.